src/App.tsx

```diff
@@ -1,14 +1,77 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { FilterOptions } from './types/FilterOptions';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+function getFilteredTodos(
+  todos: Todo[],
+  filterOption: FilterOptions,
+  query: string,
+): Todo[] {
+  const normalizedQuery = query.toLowerCase().trim();
+
+  let filteredTodos: Todo[] = todos.filter(todo => {
+    switch (filterOption) {
+      case FilterOptions.Active:
+        return !todo.completed;
+
+      case FilterOptions.Completed:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
+
+  filteredTodos = filteredTodos.filter(todo => {
+    const normalisedTodoTitle = todo.title.toLowerCase().trim();
+
+    return normalisedTodoTitle.includes(normalizedQuery);
+  });
+
+  return filteredTodos;
+}
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [filterOption, setFilterOption] = useState<FilterOptions>(
+    FilterOptions.All,
+  );
+
+  const handleSetFilterOption = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    const option = event.target.value as FilterOptions;
+
+    setFilterOption(option);
+  };
```

Comment:
This is a minor naming convention issue. The method `handleSetFilterOption` is not consistent with the naming conventions suggested in the checklist. It's recommended to use `on` prefix for the prop methods and `handle` prefix for the methods inside the component. So, in this case, it would be better to name this method `handleFilterOptionChange`.